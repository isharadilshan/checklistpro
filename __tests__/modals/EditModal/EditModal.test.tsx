import React from 'react';
import {describe, expect} from '@jest/globals';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';

import EditModal from '../../../src/components/modals/DeleteModal';

configure({adapter: new Adapter()});

const props = {
  isModalVisible: true,
  closeModal: jest.fn(),
  onEdit: jest.fn(),
};

describe('when rendering', () => {
  const wrapper = shallow(<EditModal {...props} />);

  it('should render a component matching the snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});
