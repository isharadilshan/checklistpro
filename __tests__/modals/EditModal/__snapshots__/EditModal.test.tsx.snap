// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when rendering should render a component matching the snapshot 1`] = `
<ModalWrapper
  closeModal={[MockFunction]}
  modalVisible={true}
>
  <View
    style={
      {
        "padding": 16,
      }
    }
  >
    <Title
      style={
        {
          "color": "#e76372",
          "fontSize": 24,
          "padding": 20,
          "textAlign": "center",
        }
      }
    >
      Are you sure to delete this??
    </Title>
    <Button
      label="Confirm"
      mode="contained"
      style={
        {
          "marginHorizontal": 40,
          "marginTop": 45,
        }
      }
    />
  </View>
</ModalWrapper>
`;
